### DISCUSSION TITLE
Get outputs of cpp/tensorrt_llm/kernels/cutlass_kernels/python/generate_kernels.py

### DISCUSSION THREAD

**Discussion by YLGH** [NVIDIA/TensorRT-LLM#1857] [Q&A]
Hey folks,

I couldn't find out how to run  cpp/tensorrt_llm/kernels/cutlass_kernels/python/generate_kernels.py. Does someone happen to have its outputs? I didn't have too much luck installing cutlass_library

**Comment by Advait251206** [comment]
`generate_kernels.py` is a code-generation script that depends on the CUTLASS Python code generation utilities (such as `cutlass_library`). The generated files are typically **not checked into the repository**, so you'll generally need a working CUTLASS installation to regenerate them.

If you're having trouble installing `cutlass_library`, here are a few things you can check:

- Make sure you're using the version of CUTLASS expected by the TensorRT-LLM commit you're building. The required CUTLASS revision can change between releases.
- Initialize and update submodules if you're building from source:
  ```bash
  git submodule update --init --recursive
  ```
- Ensure that the `cutlass/python` directory (or wherever `cutlass_library` resides for your CUTLASS version) is available in your `PYTHONPATH`.

If your goal is simply to build TensorRT-LLM rather than modify the kernels, it's also worth checking whether your specific release already includes the generated sources or whether your build process invokes the generator automatically.

Unfortunately, I don't have the generated outputs to share, and they can vary depending on the TensorRT-LLM version and the corresponding CUTLASS revision. If you let us know the exact TensorRT-LLM commit or release you're using, it would be easier to point you to the correct CUTLASS version or determine whether the generated files should already exist.

---

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
